### ISSUE TITLE
Breakpoint not establishing on a code line where an unititiated variable is declared

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#43787]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1009816/breakpoint-not-establishing-on-a-code-line-where-a.html)._

---
Using the Visual Studio Editor (16.5.4) , if I insert a line in my code () as follows:
      Dim x as Integer = 1
and then select the line and press <F9> to set a breakpoint it all works fine.

If I then remove the breakpoint, change the line to 
     Dim x as integer
and then select the line and press <F9> the breakpoint will not be set (nothing happens)

If I then change the line again to:
     Dim x as Integer = 1
and again set a breakpoint by press <F9> and **THEN** edit the line, amending it to:
    Dim x as Integer
the debugger remains set! (It does not vanish)

I would imagine there are many times I would want a breakpoint set on a line in which no value has been assigned to a variable. 
Is this a bug,(or a feature LOL)  and if so, can the Visual Studio Editor be updated to allow a breakpoint to be set on any line of code (except comments of course)?

---
#### Visual Studio Feedback System on 4/28/2020, 02:50 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by raisinc** [comment]
I was the originator of this feedback sent to vsfeedback

**Comment by tmat** [comment]
Not setting breakpoint without the initialization expression is intended as there is no code to execute and put breakpoint on. 

I agree having the breakpoint removed once it no longer applies would be nice.

**Comment by CyrusNajmabadi** [comment]
> I agree having the breakpoint removed once it no longer applies would be nice.

Note: the reason we don't do that is an explicit design decision.  imagine you have:

`int x = 0;`

and there's a breakpoing on that line.  you then go to edit it a bit.  you first just backspace a little to:

`int x;`

and then you add back in:

`int x = 1;`

Having the BP go away would be really unfortunate.  The downside is that you can be left with the occasional bogus BP span on some piece of syntax that won't actually apply there at runtime.

--

Note: the above issue was literally something i think i was assigned and fixed in my first couple of months on the C# team 20 years ago :D   So it's something i remember decently, unlike the thousands of other decisions in the interim time :D

**Comment by tmat** [comment]
Makes sense. We could potentially show the breakpoint as a just the red circle without a span.

**Comment by CyrusNajmabadi** [comment]
Closing out as by design.
